packages/vm/src/modules/in_memory_cache.rs

```diff
@@ -0,0 +1,81 @@
+use clru::CLruCache;
+use wasmer_runtime_core::module::Module;
+
+use crate::{Checksum, Size, VmResult};
+
+const ESTIMATED_MODULE_SIZE: Size = Size::mebi(10);
+
+/// An in-memory module cache
+pub struct InMemoryCache {
+    lru: CLruCache<Checksum, Module>,
+}
+
+impl InMemoryCache {
+    /// Creates a new cache with the given size (in bytes)
+    pub fn new(size: Size) -> Self {
+        let max_entries = size.0 / ESTIMATED_MODULE_SIZE.0;
+        InMemoryCache {
+            lru: CLruCache::new(max_entries),
+        }
+    }
+
+    pub fn store(&mut self, checksum: &Checksum, module: Module) -> VmResult<()> {
```

Comment:
Interesting, the other API uses `&Module`, this `Module`. I understand the other does not *need* the full ownership, but the same API might make this a cleaner composition?

Reply:
I tried that. LRU cache needs to own the value. When both caches take an owned module we'd have to clone it. This can be done but is not needed.